src/Microsoft.TestPlatform.BlameDataCollector/BlameCollector.cs

```diff
@@ -0,0 +1,106 @@
+        private DataCollectionEvents events;
+        private List<TestCase> TestSequence;
+        private BlameDataReaderWriter dataWriter;
+        private IBlameFileManager blameFileManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlameDataCollector"/> class.
+        /// </summary>
+        public BlameCollector()
+            : this(new XmlFileManager())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlameDataCollector"/> class.
+        /// </summary>
+        /// <param name="blameFileManager">BlameFileManager instance.</param>
+        public BlameCollector(IBlameFileManager blameFileManager)
+        {
+            this.blameFileManager = blameFileManager;
+        }
+
+        /// <summary>
+        /// Gets environment variables that should be set in the test execution environment
+        /// </summary>
+        /// <returns>Environment variables that should be set in the test execution environment</returns>
+        public IEnumerable<KeyValuePair<string, string>> GetTestExecutionEnvironmentVariables()
+        {
+            return new List<KeyValuePair<string, string>> { new KeyValuePair<string, string>("key", "value") };
+        }
+
+        /// <summary>
+        /// Initializes parameters for the new instance of the class <see cref="BlameDataCollector"/>
+        /// </summary>
+        /// <param name="configurationElement">The Xml Element to save to</param>
+        /// <param name="events">Data collection events to which methods subscribe</param>
+        /// <param name="dataSink">A data collection sink for data transfer</param>
+        /// <param name="logger">Data Collection Logger to send messages to the client </param>
+        /// <param name="environmentContext">Context of data collector environment</param>
+        public override void Initialize(XmlElement configurationElement,
+            DataCollectionEvents events, DataCollectionSink dataSink,
+            DataCollectionLogger logger, DataCollectionEnvironmentContext environmentContext)
+        {
+            ValidateArg.NotNull(logger, nameof(logger));
+            this.events = events;
+            this.dataCollectionSink = dataSink;
+            this.context = environmentContext;
+            this.logger = logger;
+            TestSequence = new List<TestCase>();
+
+            // Subscribing to events
+            this.events.SessionEnd += this.SessionEnded_Handler;
+            this.events.TestCaseStart += this.Events_TestCaseStart;
+        }
+
+        /// <summary>
+        /// Called when Test Case Start event is invoked 
+        /// </summary>
+        private void Events_TestCaseStart(object sender, TestCaseStartEventArgs e)
+        {
+            EqtTrace.Info(Constants.TestCaseStart);
+            TestCase testcase = new TestCase(e.TestElement.FullyQualifiedName, e.TestElement.ExecutorUri, e.TestElement.Source);
```

Comment:
Is this code safe?
What if one of the parameters is null? Make sure no exceptions get thrown here.
If there can be cases like that, we should use try catch.